doc/development/guide/architecture.rst

```diff
@@ -14,41 +14,91 @@ quantum circuits in a way that is compatible with classical techniques such as
 backpropagation. PennyLane thus extends the automatic differentiation
 algorithms common in optimization and machine learning to include quantum and
 :doc:`hybrid computations <glossary/hybrid_computation>`.
-A plugin system makes the framework compatible with any gate-based quantum
-simulator or hardware.
+A plugin system makes the framework compatible with many quantum
+simulators or hardware devices, remote or local.
 
-Using PennyLane, quantum computing *devices* can be used to
-evaluate quantum nodes (*QNodes*) and to return statistics of the results. To
-process the classical information obtained, *interfaces* allow using
-accelerated machine learning libraries.
+Components
+##########
 
-In most cases, computations with PennyLane are performed on a local machine
-using a simulator. Through using PennyLane plugins, one can, however, also
-utilize remote quantum devices and simulators.
+The central object in PennyLane is a **QNode**, which represents a
+"node" performing a quantum computation. Several quantum nodes may be
+part of a larger classical computation. QNodes run **quantum circuits** on **devices**.
+These devices may be simulators built into PennyLane, or external devices
+provided by plugins. The user specifies a quantum circuit by defining a "quantum function",
+which is a Python function that contains quantum gates, observables and measurements
+represented by the ``Operation``, ``Observable`` and ``MeasurementProcess`` classes, respectively).
 
-Components of PennyLane
-#######################
 
-.. image:: architecture_diagram.png
+.. image:: pl_overview.png
     :width: 800px
 
+The power of a QNode lies in the fact that it can be run in a "forward" fashion to
+execute the quantum circuit, or in a "backward" fashion in which it provides
+gradients (or, more precisely, jacobian-vector products).
```

Comment:
Would it be the case that QNodes provide vjps instead of jacobian-vector products?
```suggestion
gradients (or, more precisely, vector-jacobian products).
```